packages/gatsby-remark-images/src/index.js

```diff
@@ -83,9 +83,9 @@ module.exports = (
     const getCaptionString = () => {
       const captionOptions = Array.isArray(options.showCaptions)
         ? options.showCaptions
-        : options.showCaptions === true
-        ? [`title`, `alt`]
-        : false
+        : (options.showCaptions === true && node.alt !== EMPTY_ALT)
+          ? ["title", "alt"]
+          : false
```

Comment:
This should satisfy linter:
```suggestion
        : options.showCaptions === true && node.alt !== EMPTY_ALT
        ? [`title`, `alt`]
        : false
```

Reply:
Thanks.

Reply:
@vladar and @nategiraudeau: the linter is still not satisfied, could you please have another look?

Reply:
I'm just waiting for it to get approved.